app/src/main/java/org/schabi/newpipe/download/LoadingDialog.java

```diff
@@ -0,0 +1,90 @@
+package org.schabi.newpipe.download;
+
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.Toolbar;
+import androidx.fragment.app.DialogFragment;
+
+import org.schabi.newpipe.MainActivity;
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.databinding.DownloadLoadingDialogBinding;
+import org.schabi.newpipe.util.PermissionHelper;
+
+
+public class LoadingDialog extends DialogFragment {
+    private static final String TAG = "LoadingDialog";
+    private static final boolean DEBUG = MainActivity.DEBUG;
+    private DownloadLoadingDialogBinding dialogLoadingBinding;
+
+    @Override
+    public void onCreate(@Nullable final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (DEBUG) {
+            Log.d(TAG, "onCreate() called with: "
+                    + "savedInstanceState = [" + savedInstanceState + "]");
+        }
+        if (!PermissionHelper.checkStoragePermissions(getActivity(),
+                PermissionHelper.DOWNLOAD_DIALOG_REQUEST_CODE)) {
+            dismiss();
+            return;
+        }
```

Comment:
I don't understand why the dialog is dismissed when permissions are missing. That's already handled in the DownloadDialog.

Reply:
missed it! removing...